src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,59 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String OPERATION_SUPPLY = "supply";
+    private static final String OPERATION_BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            int[] totals = calculateTotals(fromFileName);
+            writeOutput(toFileName, totals[0], totals[1], totals[0] - totals[1]);
+        } catch (IOException e) {
+            System.err.println("An error occurred while processing the file: " + e.getMessage());
+        }
+    }
+
+    private int[] calculateTotals(String fromFileName) throws IOException {
+        int totalSupply = 0;
+        int totalBuy = 0;
+
+        try (BufferedReader br = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                String[] parts = line.split(",");
+                String operationType = parts[0].trim();
+                int amount = Integer.parseInt(parts[1].trim());
```

Comment:
Consider adding error handling for cases where the line does not contain a valid operation type or amount. This will prevent the program from crashing if the input file contains malformed lines. You can use a try-catch block around the parsing logic to handle `NumberFormatException` and `ArrayIndexOutOfBoundsException`.